### DISCUSSION TITLE
Commit history on the main page

### DISCUSSION THREAD

**Discussion by pmauduit** [allure-framework/allure2#2804] [Ideas & Feedback]
We are using allure and we are quite happy with the current features so first thanks for all the work on this project.

Something came into my mind recently, consulting the report generated for our customer: since our testsuite is versioned into a git repository, it could be interesting to have maybe the last 10 commits displayed on the main report page, so that we could see the history of added tests from the repository, as on the screenshot below:

![Screenshot from 2024-11-14 13-47-06](https://github.com/user-attachments/assets/b8de3c0f-1eef-441e-8e0e-68dcdf833809)

**Comment by shaozhb** [comment]
Can you tell me how to add SCM HISTORY to a report, which is really cool

**Reply by pmauduit** [reply]
I have not implemented anything yet, I just "hacked" the html sources of my report to showcase what I had in mind, and provide the screenshot above

**Reply by shaozhb** [reply]
That's enough for me. I'm learning to add widgets to the allure report front page, and I'm not getting anywhere.
